multicluster/values/values.go

```diff
@@ -18,22 +18,22 @@ const (
 
 // Values contains the top-level elements in the Helm charts
 type Values struct {
-	CliVersion                     string   `json:"cliVersion"`
-	ControllerImage                string   `json:"controllerImage"`
-	ControllerImageVersion         string   `json:"controllerImageVersion"`
-	Gateway                        *Gateway `json:"gateway"`
-	IdentityTrustDomain            string   `json:"identityTrustDomain"`
-	LinkerdNamespace               string   `json:"linkerdNamespace"`
-	LinkerdVersion                 string   `json:"linkerdVersion"`
-	ProxyOutboundPort              uint32   `json:"proxyOutboundPort"`
-	ServiceMirror                  bool     `json:"serviceMirror"`
-	LogLevel                       string   `json:"logLevel"`
-	ServiceMirrorRetryLimit        uint32   `json:"serviceMirrorRetryLimit"`
-	ServiceMirrorUID               int64    `json:"serviceMirrorUID"`
-	RemoteMirrorServiceAccount     bool     `json:"remoteMirrorServiceAccount"`
-	RemoteMirrorServiceAccountName string   `json:"remoteMirrorServiceAccountName"`
-	TargetClusterName              string   `json:"targetClusterName"`
-	EnablePodAntiAffinity          bool     `json:"enablePodAntiAffinity"`
+	CliVersion                      string   `json:"cliVersion"`
+	ControllerImage                 string   `json:"controllerImage"`
+	ControllerImageVersion          string   `json:"controllerImageVersion"`
+	Gateway                         *Gateway `json:"gateway"`
+	IdentityTrustDomain             string   `json:"identityTrustDomain"`
+	LinkerdNamespace                string   `json:"linkerdNamespace"`
+	LinkerdVersion                  string   `json:"linkerdVersion"`
+	ProxyOutboundPort               uint32   `json:"proxyOutboundPort"`
+	ServiceMirror                   bool     `json:"serviceMirror"`
+	LogLevel                        string   `json:"logLevel"`
+	ServiceMirrorRetryLimit         uint32   `json:"serviceMirrorRetryLimit"`
+	ServiceMirrorUID                int64    `json:"serviceMirrorUID"`
+	RemoteMirrorServiceAccount      bool     `json:"remoteMirrorServiceAccount"`
+	RemoteMirrorServiceAccountName  string   `json:"remoteMirrorServiceAccountName"`
+	TargetClusterName               string   `json:"targetClusterName"`
+	EnableTopologySpreadConstraints bool     `json:"enableTopologySpreadConstraints"`
```

Comment:
I think in order to safely remove the `EnablePodAntiAffinity` value and replace it with `EnableTopologySpreadConstraints` we'd need to go through a deprecation cycle where we temporarily support both values.  It's a bit unfortunate because both values have the same *intent* (to spread control plane pods across nodes) even if what they actually do is slightly different.

What do you think about keeping the `EnablePodAntiAffinity` value but just having it enable topology spread instead of pod anti-affinity?  This would mean the name of the value is a bit of a lie, but it avoids the need for users to migrate from one value to the other.

@alpeb do you have any thoughts on this?

Reply:
Sounds good.  Will we run into any serialization/deserialization errors during upgrade because the Values struct is changing?

Reply:
I tested this upgrading using both the CLI and Helm, and config seems to behave well. Old `enablePodAffinity` remains in the CM, but that should be fine.